File: JavaGenetiKalk/src/Monomero/nucleotide.java
package Monomero;   

public class nucleotide extends Monomero{
    private boolean _hasOxy;
    private String _comp;

    public nucleotide(String n, String s, double p, boolean hasOxy, String comp){
        super(n, s, p);
        _hasOxy=hasOxy;
        _comp=comp;
    }

    public String getComp(){
        return _comp;
    }

    public boolean getHasOxy(){
        return _hasOxy;
    }

    public void setComp(String comp){
        _comp=comp;
    }

    public void setHasOxy(boolean hasOxy){
        _hasOxy=hasOxy;
    }

}

File: GENETIKalk_FINAL/model/nucleotide.cpp
//
// Created by andre on 10/01/2018.
//
#include "nucleotide.h"

//Ridefinizione di operator<<
std::ostream& operator<<(std::ostream& os, const Nucleotide& n) {
    return os << n.getName().toUtf8().constData() << " - " << n.getAcr().toUtf8().constData();
}

//Costruttore
Nucleotide::Nucleotide(QString n, QString s, double p, bool h, QString co) : Monomero(std::move(n), s, p), _hasOxy(h), _comp(co){ }

//Metodi di get
QString Nucleotide::getComp() const {
    return _comp;
}

bool Nucleotide::getHasOxy() const {
    return _hasOxy;
}

//Metodi di set
void Nucleotide::setComp(const QString & c) {
    _comp = c;
}

void Nucleotide::setHasOxy(const bool & h) {
    _hasOxy = h;
}

Nucleotide* Nucleotide::clone() const {
    return new Nucleotide(*this);
}

//Operatori ridefiniti
bool Nucleotide::operator==(const Nucleotide& n) const {
    return getName() == n.getName() && getAcr() == n.getAcr() && getHasOxy() == n.getHasOxy();
}



